src/main/scala/com/vngrs/etl/Combine.scala

```diff
@@ -0,0 +1,49 @@
+package com.vngrs.etl
+
+import org.apache.spark.rdd.RDD
+
+import scala.reflect.ClassTag
+
+/**
+  * [[com.vngrs.etl.Combine]] is an interface for combining two [[com.vngrs.etl.Pipeline]]s into one.
+  *
+  * @tparam A Type of Input [[com.vngrs.etl.Pipeline]]
+  * @tparam B Type of Output [[com.vngrs.etl.Pipeline]]
+  */
+trait Combine[A, B] {
```

Comment:
There are 3 problems with design here

* 2 implementations, zip, and union, have non-generic method along with generic one, combine, so it is not generic not custom.
* Number of implementations are limited to: zip, union, join, leftJoin, rightJoin, and outerJoin
* It is impossible to understand implementations of this interface handles with 2 pipelines.